src/ngx_http_lua_output.c

```diff
@@ -802,4 +802,99 @@ ngx_http_lua_flush_cleanup(void *data)
     ctx->flushing_coros--;
 }
 
+#ifndef NGX_LUA_NO_FFI_API
+//Pre-Condition: offset + len <= size of buf
+int
+ngx_http_lua_ffi_write(ngx_http_request_t *r, const char *buf, size_t offset,
+                       size_t len, char **err)
+{
+    ngx_http_lua_ctx_t          *ctx;
+    ngx_buf_t                   *b;
+    ngx_chain_t                 *cl;
+    ngx_int_t                    rc;
+    const char                  *str_sub;
```

Comment:
The `str_sub` variable name is confusing to me. Are you suggesting the Lua function name `string.sub()` here? I think you can just use the variable name `p` here.